HWs/Test3/task2/main.c

```diff
@@ -0,0 +1,107 @@
+#include <stdio.h>
+#include "mergeSortingList.h"
+
+#define maximumSize 256
+
+void printError()
+{
+    printf("Error codes:\n");
+    printf("-1: File opening error\n");
+    printf("-2: Memory allocation error\n");
+    printf("-3: Incorrect file structure\n");
+}
+
+Error fillList(List **list, char *fileName)
+{
+    FILE *file = fopen(fileName, "r");
+    if (file == NULL || feof(file))
+    {
+        return -1;
+    }
+
+    char name[maximumSize] = {0};
+    int amount = 0;
+    char currentCharacter = 0;
+    int counter = 1;
+    int errorCode = 0;
+    int scanResult = 0;
+    name[0] = '\n'; // Preservation of the invariant "Name starts with '\n'"
+    while (!feof(file))
+    {
+        while ((currentCharacter = getc(file)) != '-' && !feof(file))
+        {
+            name[counter] = currentCharacter;
+            ++counter;
+        }
+        if (feof(file))
+        {
+            return -3;
+        }
+        name[counter] = '\0';
+        scanResult = fscanf(file, "%d", &amount);
+        if (scanResult == 0)
+        {
+            return -3;
+        }
+        if ((errorCode = addElement(list, name, amount)) != 0)
+        {
+            return errorCode;
+        }
+
+        for (int i = 0; i < counter; ++i)
+        {
+            name[i] = 0;
+        }
+        counter = 0;
+    }
+
+    fclose(file);
+    return 0;
+}
+
+int main()
+{
+    List *list = NULL;
+    int errorCode = fillList(&list, "../HWs/Test3/task2/storage.txt");
+    if (errorCode != 0)
+    {
+        printf("The fillList function failed with error %d.\n", errorCode);
+        printError();
+        return -1;
+    }
+    int n = 0;
+    printf("Enter the number \"N\": ");
```

Comment:
```suggestion
    printf("Enter amount of products with 50% discount: ");
```
Сообщения всегда должны быть информативными.
(-0.25)